code/game/objects/pool/pool.dm

```diff
@@ -0,0 +1,308 @@
+//Simple pool behaviour. Sprites by Cdey!
+
+/**
+How to pool!
+Place pool turfs on the inside of your pool. This is where you swim. Pool end caps to cap it off on the north face
+Place pool border decals around the pool so it doesn't look weird
+Place a pool ladder at the top of the pool so that it leads to a normal tile (or else it'll be hard to get out of the pool.)
+Place a pool filter somewhere in the pool if you want people to be able to modify the pool's settings (Temperature) or dump reagents into the pool (which'll change the pool's colour)
+*/
+
+GLOBAL_LIST_EMPTY(species_swimming_components)
+
+/obj/effect/overlay/poolwater
+	name = "Pool water"
+	icon = 'icons/obj/pool.dmi'
+	icon_state = "water"
+	anchored = TRUE
+	layer = ABOVE_ALL_MOB_LAYER
+	mouse_opacity = MOUSE_OPACITY_TRANSPARENT
+
+/turf/open/indestructible/sound/pool
+	name = "Swimming pool"
+	desc = "A fun place where you go to swim! <b>Drag and drop yourself onto it to climb in...</b>"
+	icon = 'icons/obj/pool.dmi'
+	icon_state = "pool"
+	sound = 'sound/effects/splash.ogg'
+	var/id = null //Set me if you don't want the pool and the pump to be in the same area, or you have multiple pools per area.
+	var/obj/effect/water_overlay = null
+
+/turf/open/indestructible/sound/pool/end
+	icon_state = "poolwall"
+
+/turf/open/indestructible/sound/pool/Initialize(mapload)
+	. = ..()
+	water_overlay = new /obj/effect/overlay/poolwater(get_turf(src))
+	if(!LAZYLEN(GLOB.species_swimming_components))
+		//Setup the swimming species
+		GLOB.species_swimming_components[/datum/species/squid] = /datum/component/swimming/squid
+		GLOB.species_swimming_components[/datum/species/ethereal] = /datum/component/swimming/ethereal
+		GLOB.species_swimming_components[/datum/species/jelly] = /datum/component/swimming/disolve
+		for(var/golem_type in typesof(/datum/species/golem))
+			GLOB.species_swimming_components[golem_type] = /datum/component/swimming/golem
```

Comment:
I really don't like this, would be much better with some kind of proc like get_swimming_datum that returns the swimming datum based on the passed in species through a switch, since this is storing the same datum for every single golem type there is.
```
/proc/get_swimming_component(species)
   switch(species)
      if(datum/species/squid)
         return /datum/component/swimming/squid
   etc.
```